tao_compiler/mlir/xla/ral/BUILD

```diff
@@ -273,6 +274,7 @@ cc_library(
         "context/common_context_impl.h",
     ] + if_cuda_or_rocm([
         "context/stream_executor_based_impl.h",
+        "context/common_context_impl_sparse.h",
```

Comment:
maybe use context/cuda_sparse_impl.[h|cc] is more precise?

Reply:
We may add more sparse kernels to this file in the future, maybe implementing them in one file is better than separating them into different files?

Reply:
it's under `if_cuda_or_rocm` condition in BUILD file. so I think it's gpu only?

Reply:
I think spgemm is not implemented for rocm yet, so it should be  `if_cuda` instead of `if_cuda_or_rocm`.

Reply:
Done